src/components/OpenWith/OpenWith.js

```diff
@@ -86,16 +102,96 @@ class OpenWith extends PureComponent<Props> {
     }
 
     /**
-     * Fetches the list of potential integrations on load
      *
      * @private
      * @inheritdoc
      * @return {void}
      */
     componentDidMount() {
-        /* no-op */
+        const { fileId }: Props = this.props;
+        this.api.getOpenWithAPI(false).get({
+            id: fileId,
+            successCallback: this.fetchAppIntegrations,
+            errorCallback: this.fetchErrorCallback
+        });
     }
 
+    /**
+     * Fetch app intgrations info needed to render.
+     *
+     * @param {OpenWithIntegrations} openWithintegrations - The available Open With integrations
+     * @return {void}
+     */
+    fetchAppIntegrations = (openWithintegrations: OpenWithIntegrations) => {
+        const { items } = openWithintegrations;
+        const appIntegrationInfoPromises = [];
+
+        items.forEach((integrationItem: OpenWithIntegrationItem) => {
```

Comment:
Consider using `items.map` to create the promises array:

```
const appIntegrationInfoPromises = items.map(({ app_integration: { id }}) => (
    this.fetchAppIntegrationPromise(id)
);

Reply:
this + async await makes it much much cleaner